### ISSUE TITLE
Relax mod in Lazer has improper hit window compared to stable

### ISSUE THREAD

**Issue by Remyria** [ppy/osu#31134]
### Type

Game behaviour

### Bug description

There is a significant discrepancy between stable and lazer on how relax works.

On stable, relax hits a bit earlier than perfect timing (12ms) to give a bit of margin if you aim a tiny bit too fast. However, on lazer, it's almost EXACTLY on the perfect timing(only 3ms), which makes using the mod rather annoying most of the time.

I do play a reasonable amount of relax and at least to me, the issue is very noticeable when I get misses that I wouldn't even have gotten playing with no mod due to how strict it currently is for early hits.

It may not seem like a big difference, by a factor of 4 is the same as between OD0 and OD10, or OD11.33 and OD12.83 in this case.

### Screenshots or videos

First image: Hit window on Stable at OD10
![image](https://github.com/user-attachments/assets/4be6919b-380b-4531-9022-8c7b005ba704)

Second image: Hit window on Lazer at OD10
![image](https://github.com/user-attachments/assets/b528b9b4-82a9-4c51-9685-ae7fd4309a4e)

You can see how big the difference is on the acc bar

EDIT: checked, and where lazer hits 3ms early, stable(info from mcosu) hits 12ms early. meaning early aim error margin on stable is such as hitting a 300 on OD11.33 and on lazer OD12.83 with both having whatever the map's OD is for late aim error margin, meaning you generally want to hit way late.
[Also rewrote the description slightly].

### Version

2024.1208.0

### Logs

Inapplicable

**Comment by Paladynee** [comment]
It makes sense that relax hits just on time. I think we should unlearn the early hit margin of relax, since it also damages your real gameplay as well. While you don't consistently hit the note at just the right time, we don't hit the note consistently early either. I think this is an intended change, and it should stay that way.

Trivia: there was a much bigger discrepancy in stable relax vs lazer relax, the issue being that stable does not check whether the object that's going to be clicked by relax autoclick (obj1) is not checked if it is the same object that is currently aimed (obj2). If these 2 objects are the same object, it plays completely fine, but however, if obj2 is a different object than obj1, relax just spams the button, causing either a notelock vibration effect or a sliderhead miss on stable.

TL;DR: try aiming at different objects than the one you should've aimed for. you'll see that relax will try to press them, causing you to miss.

**Comment by Remyria** [comment]
I fundamentally disagree.

I think Lazer's relax is both unfun(right now) and a bigger issue for your real gameplay then stable's because as it stands, it encourages you to aim consistently late, because if you're slightly early(still well within 300 hit window) when you leave the circle, which happens roughly half the time nomod, you will miss.

On the flip side, I don't think stable's relax teaches you to aim too early at all, it's merely accounting for your aim instability by giving you a few extra milliseconds that you can be early by. we're talking around 10ms early on stable, which is enough to feel good and hit stuff, but not enough to change how you aim compared to nomod

**Comment by minisbett** [comment]
This is not a bug, it's an intended change for relax on Lazer. But I'd also love to see it get reverted, as someone who used to play a lot of relax, Lazer's relax doesn't feel relaxing anymore.

**Comment by bdach** [comment]
> This is not a bug, it's an intended change for relax on Lazer

[citation needed]

**Comment by minisbett** [comment]
> > This is not a bug, it's an intended change for relax on Lazer
> 
> [citation needed]
> 
> i don't recall anything like that being stated anywhere

I've talked to tsunyoku about Relax many times and what he said makes me understand that whilst it isn't a logical change it is something done with awareness.
![Screenshot_2024-12-26-09-24-39-527_com.discord-edit.jpg](https://github.com/user-attachments/assets/b214ef3c-8baa-443f-b681-dbf557c3607e)

**Comment by bdach** [comment]
> I've talked to tsunyoku about Relax many times and what he said makes me understand that whilst it isn't a reasoned change it is something done with awarenes

all this is news to me

- relax was contributed by an external contributor ages ago before there were many eyes on lazer https://github.com/ppy/osu/pull/3172
- "weren't discussed" implies *there wasn't really any serious discussion on it*
- the discussion thread you linked has only one response from me that goes basically "sure we can change it why not"

so please, before putting out statements like "intended change for relax on lazer" i'd appreciate double checking that you're not putting words in our mouths

**Comment by minisbett** [comment]
I'm just saying, calling this improper doesn't feel right because it was implemented differently with awareness. There was no intention to have it match stable.

My wording with it being an intended change was confusing, what I simply meant is that it's not a bug because it's not expected to match stable.

**Comment by tsunyoku** [comment]
That is a wild way to interpret a message which says nothing other than "it is weird". bdach is right - osu!(lazer)'s relax mod was contributed by someone outside of the team and has been discussed by basically nobody since.

If you're going to pull out random Discord messages and use them to quote me, at least don't put words in my mouth. I don't appreciate that.

Apart from that, I might as well leave my own opinions about the leniency itself. I personally agree that the leniency should be changed to match stable for a few reasons:

- All PP balancing is done with the knowledge and understanding of relax plays set on osu!stable
- I think it makes most sense for mods to match osu!stable unless there is a good reason for it not to, which I think doesn't apply here
- For the average person playing relax on osu!(lazer), they're probably not aware of the leniency being different and it may play weirdly

However, changing something core to this mod is probably going to require score invalidations? I'm not sure we've done this on a mod-scale yet.

**Comment by bdach** [comment]
> changing something core to this mod is probably going to require score invalidations

It's an unranked mod with 0.1x multiplier so eh, it's not like there'll be leaderboard impact (RX leaderboards don't even really exist now). Can probably mark plays as unranked and call it a day, not sure anybody will be mad. Or just ignore it, if we're making relax easier then the harder plays may as well stay (contrary to what would have to happen if it were to be made harder).
